skimage/morphology/misc.py

```diff
@@ -56,21 +57,38 @@ def _check_dtype_supported(ar):
         )
 
 
-def remove_small_objects(ar, min_size=64, connectivity=1, *, out=None):
+@deprecate_parameter(
+    deprecated_name="min_size",
+    new_name="max_size",
+    start_version="0.26.0",
+    stop_version="2.0.0",
+    template=f"{deprecate_parameter.replace_parameter_template} "
+    "Note that the new threshold removes objects smaller than **or equal to** "
+    "its value, while the previous parameter only removed smaller ones.",
+)
+def remove_small_objects(
+    ar, min_size=DEPRECATED, connectivity=1, *, max_size=64, out=None
+):
     """Remove objects smaller than the specified size.
 
-    Expects ar to be an array with labeled objects, and removes objects
-    smaller than min_size. If `ar` is bool, the image is first labeled.
-    This leads to potentially different behavior for bool and 0-and-1
+    Expects `ar` to be an array with labeled objects, and removes objects
+    smaller than `max_size`. If `ar` is bool, the image is first labeled.
+    This leads to potentially different behavior for bool vs. 0-and-1
     arrays.
 
     Parameters
     ----------
     ar : ndarray (arbitrary shape, int or bool type)
         The array containing the objects of interest. If the array type is
         int, the ints must be non-negative.
-    min_size : int, optional (default: 64)
-        The smallest allowable object size.
+    max_size : int, optional (default: 64)
+        Remove objects whose contiguous area (or volume in N-D) contains this
+        number of pixels or fewer.
+
+        .. versionchanged:: 0.26
+            Replaces deprecated `min_size` which only removed objects strictly
+            smaller than its size.
```

Comment:
Maybe we should say why? I.e., to make the naming less confusing.

Reply:
I gave it a try in https://github.com/scikit-image/scikit-image/pull/7739/commits/50366c7bc99893c3c395e3e4b6aa33502dce5268. Feel welcome to tweak further if you want. :)